pkg/api/store/cluster/cluster_store.go

```diff
@@ -434,3 +434,28 @@ func setPrivateRegistryPasswordIfNotExists(oldData, newData map[string]interface
 	}
 	values.PutValue(newData, updatedConfig, "rancherKubernetesEngineConfig", "privateRegistries")
 }
+
+func setCloudProviderPasswordFields(oldData, newData map[string]interface{}) {
```

Comment:
we need to call this in store update, https://github.com/rancher/rancher/blob/e2342788a35de38a9597c5b7924b822a8b36f15c/pkg/api/store/cluster/cluster_store.go#L270

Reply:
yup I still haven't covered all fields in it, so I put it as a draft PR :)

Reply:
Didn't realize it was draft PR, sorry!